modules/core/src/matrix_wrap.cpp

```diff
@@ -38,25 +52,91 @@ Mat _InputArray::getMat_(int i) const
         return Mat(sz, flags, obj);
     }
 
-    if( k == STD_VECTOR )
+    if( k == STD_VECTOR || k == STD_VECTOR_VECTOR )
     {
-        CV_Assert( i < 0 );
-        int t = CV_MAT_TYPE(flags);
-        const std::vector<uchar>& v = *(const std::vector<uchar>*)obj;
-        int v_size = size().width;
+        int typ = CV_MAT_TYPE(flags);
+        int esz = CV_ELEM_SIZE(typ);
+
+        CV_Assert((k == STD_VECTOR && i < 0) ||
+                  (k == STD_VECTOR_VECTOR && i >= 0));
+
+        #undef GET_VEC_MAT
+        #define GET_VEC_MAT(T) \
+        if (k == STD_VECTOR) { \
+            const std::vector<T >* v = (const std::vector<T >*)obj; \
+            CV_Assert( v->size() <= (size_t)INT_MAX ); \
+            int ni = (int)v->size(); \
+            return ni > 0 ? Mat(1, &ni, typ, (void*)v->data()) : Mat(); \
+        } else { \
+            const std::vector<std::vector<T > >* vv = (const std::vector<std::vector<T > >*)obj; \
+            CV_Assert( (size_t)i < vv->size() ); \
+            const std::vector<T >& vi = vv->at(i); \
+            CV_Assert( vi.size() <= (size_t)INT_MAX ); \
+            int ni = (int)vi.size(); \
+            return ni > 0 ? Mat(1, &ni, typ, (void*)vi.data()) : Mat(); \
+        }
 
-        return !v.empty() ? Mat(1, &v_size, t, (void*)&v[0]) : Mat();
+        switch( esz )
```

Comment:
What about std::vector<std::vector<bool>>? We have such thing as correspondence matrix in multi-camera calibration. Also it could be used as points mask.

Reply:
STD_VECTOR_VECTOR is handled properly in this PR. If you are talking about `vector<vector<vector<T>>>`, that's out of scope of this PR